safe_transaction_service/history/indexers/ethereum_indexer.py

```diff
@@ -145,8 +145,16 @@ def get_block_numbers_for_search(
         if (from_block_number + self.block_process_limit) >= (
             current_block_number - self.confirmations
         ):
+            reindex_blocks = self.blocks_to_reindex_again
+            if self.blocks_to_reindex_again >= self.block_process_limit:
```

Comment:
Please add tests for this

Reply:
Also I'm a little worried that if `block_process_limit` is too tight we can miss reindexing some transactions, but I think there's no easy solution for that